From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct rq
Patch-mainline: never, kabi

In 3.12.56, commit 69836a46b1c986665c514106b0137984cacb18dc (sched:
Replace post_schedule with a balance callback list), upstream commit
e3fca9e7cbfb72694a21c886fcdf9f059cfded9c added and removed one member
from/to struct rq. It indeed changed the layout of the structure.

But given this is only scheduler-internal structure, protect the
members the usual way, by #ifdef __GENKSYMS__.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 kernel/sched/sched.h |    5 +++++
 1 file changed, 5 insertions(+)

--- a/kernel/sched/sched.h
+++ b/kernel/sched/sched.h
@@ -482,10 +482,15 @@ struct rq {
 
 	unsigned long cpu_power;
 
+#ifndef __GENKSYMS__
 	struct callback_head *balance_callback;
+#endif
 
 	unsigned char idle_balance;
 	/* For active balancing */
+#ifdef __GENKSYMS__
+	int post_schedule;
+#endif
 	int active_balance;
 	int push_cpu;
 	struct cpu_stop_work active_balance_work;
